ui/apps/platform/src/Containers/NetworkGraph/utils/flowUtils.ts

```diff
@@ -19,3 +32,120 @@ export function getNumFlows(flows: Flow[]): number {
+        if ((edge.source !== id && edge.target !== id) || !edge.source || !edge.target) {
+            return acc;
+        }
+        const adjacentNodeId = edge.source !== id ? edge.source : edge.target;
+        const adjacentNode = controller.getNodeById(adjacentNodeId);
+        const adjacentNodeData: CustomNodeData = adjacentNode?.getData();
+        const result = edge.data.properties.map(({ port, protocol }): Flow => {
+            const direction: string = edge.source === id ? 'Egress' : 'Ingress';
+            const type = adjacentNodeData.type === 'DEPLOYMENT' ? 'Deployment' : 'External';
+            let entity = '';
+            let namespace = '';
+            if (adjacentNodeData.type === 'DEPLOYMENT') {
+                entity = adjacentNodeData.deployment.name;
+                namespace = adjacentNodeData.deployment.namespace;
+            } else if (adjacentNodeData.type === 'CIDR_BLOCK') {
+                entity = `${adjacentNodeData.externalSource.name}`;
+            } else if (adjacentNodeData.type === 'EXTERNAL_ENTITIES') {
+                entity = 'External entities';
+            }
+            // we need a unique id for each network flow
+            const flowId = `${entity}-${namespace}-${direction}-${port}-${protocol}`;
+            return {
+                id: flowId,
+                type,
+                entity,
+                namespace,
+                direction,
+                port: String(port),
+                protocol: protocolLabel[protocol],
+                // @TODO: Need to set this depending on whether it is in the baseline or not
+                isAnomalous: true,
+                // @TODO: Need to create nesting structure
+                children: [],
+            };
+        });
+        return [...acc, ...result] as Flow[];
+    }, [] as Flow[]);
+    return networkFlows;
+}
+
+/*
+  This function takes network flows and filters the data based on a text search value 
+  for entity name and some advanced filters specific to network flows. These include:
+  flow types, directionality, protocols, and ports
+*/
+export function filterNetworkFlows(
+    flows: Flow[],
+    entityNameFilter: string,
+    advancedFilters: AdvancedFlowsFilterType
+): Flow[] {
+    const filteredFlows = flows.filter((flow) => {
+        let matchedEntityName = false;
+        let matchedFlowType = true;
+        let matchedDirectionality = true;
+        const matchedProtocol = true;
+        let matchedPort = true;
+
+        // check filtering by entity name
+        if (flow.entity.includes(entityNameFilter)) {
+            matchedEntityName = true;
+        }
+
+        // check filtering by flow type
+        if (advancedFilters.flows.length) {
+            const isAnomalousFiltered =
+                advancedFilters.flows.includes('anomalous') && flow.isAnomalous;
+            const isBaselineFiltered =
+                advancedFilters.flows.includes('baseline') && !flow.isAnomalous;
```

Comment:
isn't it either anomalous or baseline filtered? i feel like it makes more sense to do this conditionally if so

Reply:
I guess I'm not understanding what you mean. It's true a network flow is either anomalous or baseline, but with the filters, we can select both `Anomalous` and `Baseline`. So we show anomalous network flows when (1) No filter is selected (2) The anomalous filter is selected (3) When the anomalous filter is selected with the baseline filter

Reply:
The filtering works this way. If there's a different way you think would be more ideal, and gets the same result, we can try that. I was just confused about what you meant by "it makes more sense to do this conditionally"

Reply:
I'll merge for now and we can discuss